core/git_command.py

```diff
@@ -268,42 +270,41 @@ def find_git_toplevel(self, folder, throw_on_stderr):
 
     @property
     def repo_path(self):
-        return self._repo_path()
-
-    @property
-    def short_repo_path(self):
-        if "HOME" in os.environ:
-            return self.repo_path.replace(os.environ["HOME"], "~")
-        else:
-            return self.repo_path
-
-    def _repo_path(self, throw_on_stderr=True):
         """
         Return the absolute path to the git repo that contains the file that this
         view interacts with.  Like `file_path`, this can be overridden by setting
         the view's `git_savvy.repo_path` setting.
-
-        Do not raise error when throw_on_stderr is `False`. It is needed
-        in GsUpdateStatusBarCommand, otherwise, spurious popup will be shown.
         """
         # The below condition will be true if run from a WindowCommand and false
         # from a TextCommand.
         view = self.window.active_view() if hasattr(self, "window") else self.view
         repo_path = view.settings().get("git_savvy.repo_path") if view else None
 
         if not repo_path or not os.path.exists(repo_path):
-            repo_path = self.find_repo_path(throw_on_stderr=throw_on_stderr)
+            repo_path = self.find_repo_path()
             if not repo_path:
-                if throw_on_stderr:
-                    raise ValueError("Unable to determine Git repo path.")
+                window = view.window()
+                if window and window.folders():
+                    raise ValueError("Not a git repository.")
                 else:
-                    return None
-            else:
-                view.settings().set("git_savvy.repo_path", os.path.realpath(repo_path))
+                    raise ValueError("Unable to determine Git repo path.")
+
+            if view:
+                file_name = view.file_name()
+                # only set "git_savvy.repo_path" when the current file is in repo_path
+                if file_name and os.path.realpath(file_name).startswith(repo_path + os.path.sep):
+                    view.settings().set("git_savvy.repo_path", repo_path)
 
         return os.path.realpath(repo_path) if repo_path else repo_path
 
     @property
+    def short_repo_path(self):
+        if "HOME" in os.environ:
+            return self.repo_path.replace(os.environ["HOME"], "~")
```

Comment:
Is this necessary for some cases? I don't recall an  issue discussing it. Maybe, note the reason in a comment / docstring?